packages/mobile/src/identity/contactMapping.ts

```diff
@@ -218,19 +221,44 @@ export function* fetchAddressesAndValidateSaga({
   }
 }
 
-function* getAddresses(e164Number: string) {
-  let phoneHash: string
-  if (features.USE_PHONE_NUMBER_PRIVACY) {
-    const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
-    phoneHash = phoneHashDetails.phoneHash
-  } else {
-    phoneHash = getPhoneHash(e164Number)
-  }
-
+function* getAccountAddresses(e164Number: string) {
+  const phoneHashDetails: PhoneNumberHashDetails = yield call(fetchPhoneHashPrivate, e164Number)
+  const phoneHash = phoneHashDetails.phoneHash
   const lookupResult: IdentifierLookupResult = yield call(lookupAttestationIdentifiers, [phoneHash])
   return getAddressesFromLookupResult(lookupResult, phoneHash) || []
 }
 
+function* getWalletAddressesAndUpdateCache(e164Number: string) {
+  const contractKit = yield call(getContractKit)
+  const accountsWrapper: AccountsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAccounts,
+  ])
+
+  const accountAddresses: string[] = yield call(getAccountAddresses, e164Number)
+  const walletAddresses: string[] = yield all(
+    accountAddresses.map((accountAddress) => call(accountsWrapper.getWalletAddress, accountAddress))
+  )
+
+  const possibleUserAddresses: string[] = []
+  const walletToAccountAddress: WalletToAccountAddressType = {}
+  walletAddresses.forEach((address, i) => {
+    const walletAddress = address.toLowerCase()
+    const accountAddress = accountAddresses[i]
+    // `getWalletAddress` returns 0x0 when there isn't a wallet registered
+    if (walletAddress !== '0x0') {
+      walletToAccountAddress[walletAddress] = accountAddress
+      possibleUserAddresses.push(walletAddress)
+    } else {
+      // NOTE: Only need this else block if we are not confident all wallets are registered
+      walletToAccountAddress[accountAddress] = accountAddress
```

Comment:
I'm not sure I understand the intention of this mapping. When is it meaningful to say an account address maps to itself? What's a wallet in this context if not one of the `walletAddresses` from above? Can you help me understand this bit?

Reply:
Adding to the mapping in this block is mostly for completeness rather than absolute necessity.

In the `doFetchDataEncryptionKey` saga, I need a way to go from EOA to MTW. I need this because moving forward the DEK will be registered to the MTW and the metadata included in transactions (for encryption) only includes the EOA.

Reply:
Thanks for clarifying. I think leaving the mapping undefined is slightly more clear but I'll leave that up to you :)